code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -220,3 +220,19 @@
 
 /mob/living/carbon/human/get_biological_state()
 	return dna.species.get_biological_state()
+
+///Returns death message for mob examine text
+/mob/living/carbon/human/proc/generate_death_examine_text()
+	var/mob/dead/observer/ghost = get_ghost(TRUE, TRUE)
+	var/t_He = p_they(TRUE)
+	var/t_his = p_their()
+	var/t_is = p_are()
+	if(getorgan(/obj/item/organ/brain) && !key) //The death mob has a brain and no client/player that is assigned to the mob
+		if (!ghost || !ghost.can_reenter_corpse)  //And there is no ghost that could reenter the body
+			//There is no way this mob can in any normal way get a player, so he lost the will to live
+			return "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life and [t_his] soul has lost the will to live...</span>"
+		else //There is a ghost and it can reenter
+			//This mob has a ghost linked that could still reenter the body, so the soul only departed
+			return "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life and [t_his] soul has departed...</span>"
+	//If no special cases matches return the default death message
+	return "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life...</span>"
```

Comment:
```suggestion
	if(!getorgan(/obj/item/organ/brain) || key)
		return "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life...</span>" //Default death message
	//The death mob has a brain and no client/player that is assigned to the mob
	if(!ghost?.can_reenter_corpse)  //And there is no ghost that could reenter the body
		//There is no way this mob can in any normal way get a player, so they lost the will to live
		return "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life and [t_his] soul has lost the will to live...</span>"
	//This mob has a ghost linked that could still reenter the body, so the soul only departed
	return "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life and [t_his] soul has departed...</span>"
```

Reply:
Won't this change the logic when there is no ghost?
Like for example someone spectated (so the dead mob now has no key) and joined again via some ghost spawn.
`!ghost?.can_reenter_corpse` would now be null so false and it would reach the last return instead of the "lost hiss will to live" return which should normally be triggered since there is nobody who can reenter the body.
I guess the easiest would be to change line 233 to `isnull(ghost) || !ghost.can_reenter_corpse`?

Reply:
`!(null?.can_reenter_corpse)` -> `!(null)` -> `TRUE`
The `?.` operator gets resolved before the `!` negation one, and doesn't skip it.

If you wanted the behavior you describe you'd have to write it as `if(ghost && !ghost.can_reenter_corpse)`